xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -7258,7 +7239,7 @@ glusterd_get_brickinfo(xlator_t *this, const char *brickname, int port,
         {
             if (gf_uuid_compare(tmpbrkinfo->uuid, MY_UUID))
                 continue;
-            if (!strcmp(tmpbrkinfo->path, brickname) &&
+            if (!strcmp(tmpbrkinfo->path, brickname) ||
```

Comment:
not sure why is this change done here, but in any case, perhaps worth doing the port comparison before the strcmp()

Reply:
To incorporate the functions glusterd_pmap_portbybrick and glusterd_pmap_bruckbyport to work fine.
Will, make the changes to compare port first.